File: CS/DocumentIteratorExample/Form1.cs
using DevExpress.XtraRichEdit.API.Native;
using System;
using System.Linq;

namespace DocumentIteratorExample {
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm {
        public Form1() {
            InitializeComponent();
            ribbonControl1.SelectedPage = pageIterator;
            richEditControl1.LoadDocument("Markdown.docx");                 
        }

        private void btnIteratorRun_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            MarkdownVisitor visitor = new MarkdownVisitor();
            Iterate(visitor);
           
            memoEdit1.Text = visitor.Text;
        }
        private void Iterate(IDocumentVisitor visitor)
        {
            DocumentIterator iterator = new DocumentIterator(richEditControl1.Document, true);
            while (iterator.MoveNext())
                iterator.Current.Accept(visitor);
        }
    }

}

File: VB/DocumentIteratorExample/Form1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraRichEdit.API.Native
Imports System
Imports System.Linq

Namespace DocumentIteratorExample
	Partial Public Class Form1
		Inherits DevExpress.XtraBars.Ribbon.RibbonForm
		Public Sub New()
			InitializeComponent()
			ribbonControl1.SelectedPage = pageIterator
			richEditControl1.LoadDocument("Markdown.docx")
		End Sub

		Private Sub btnIteratorRun_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles btnIteratorRun.ItemClick
			Dim visitor As New MarkdownVisitor()
			Iterate(visitor)

			memoEdit1.Text = visitor.Text
		End Sub
		Private Sub Iterate(ByVal visitor As IDocumentVisitor)
			Dim iterator As New DocumentIterator(richEditControl1.Document, True)
			Do While iterator.MoveNext()
				iterator.Current.Accept(visitor)
			Loop
		End Sub
	End Class

End Namespace
